server/config/client.go

```diff
@@ -81,9 +81,18 @@ func GenerateClientConfig(c *model.Config, telemetryID string, license *model.Li
 	props["EnableGifPicker"] = strconv.FormatBool(*c.ServiceSettings.EnableGifPicker)
 	props["GfycatApiKey"] = *c.ServiceSettings.GfycatAPIKey
 	props["GfycatApiSecret"] = *c.ServiceSettings.GfycatAPISecret
-	props["GiphySdkKey"] = *c.ServiceSettings.GiphySdkKey
 	props["MaxFileSize"] = strconv.FormatInt(*c.FileSettings.MaxFileSize, 10)
 
+	if model.GetServiceEnvironment() == model.ServiceEnvironmentProduction {
+		if *c.ServiceSettings.GiphySdkKey != "" {
+			props["GiphySdkKey"] = *c.ServiceSettings.GiphySdkKey
+		} else {
+			props["GiphySdkKey"] = model.MattermostGiphySdkKey
+		}
+	} else if model.GetServiceEnvironment() == model.ServiceEnvironmentDev || model.GetServiceEnvironment() == model.ServiceEnvironmentTest {
+		props["GiphySdkKey"] = model.ServiceSettingsDefaultGiphySdkKeyTest
+	}
```

Comment:
Let's move this to a separate function to keep things clear at a high level.

```
props["GiphySdkKey"] = getGiphyKey(c.ServiceSettings)
```

Now you can also add a unit test :)

Reply:
Sure will do that

Reply:
added the function and the test